DXRVanilla/DeusEx/Classes/Player.uc

```diff
@@ -861,9 +861,83 @@ function bool IsFrobbable(actor A)
 
 function HighlightCenterObject()
 {
-    local Vector loc;
+    HighlightCenterObjectMain();
+    HighlightCenterObjectLaser();
+}
+
+function HighlightCenterObjectMain()
+{
+    local Actor target, smallestTarget;
+    local Vector HitLoc, HitNormal, StartTrace, EndTrace;
+    local DeusExRootWindow root;
+    local float minSize;
+    local bool bFirstTarget;
+
+    if (IsInState('Dying'))
+        return;
+
+    root = DeusExRootWindow(rootWindow);
+
+    // we do the trace every frame, unlike the vanilla behaviour of doing it every 100ms
+
+    // figure out how far ahead we should trace
+    StartTrace = Location;
+    EndTrace = Location + (Vector(ViewRotation) * MaxFrobDistance);
+
+    // adjust for the eye height
+    StartTrace.Z += BaseEyeHeight;
+    EndTrace.Z += BaseEyeHeight;
+
+    smallestTarget = None;
+    minSize = 99999;
+    bFirstTarget = True;
+
+    // find the object that we are looking at
+    // make sure we don't select the object that we're carrying
+    // use the last traced object as the target...this will handle
+    // smaller items under larger items for example
+    // ScriptedPawns always have precedence, though
+    foreach TraceActors(class'Actor', target, HitLoc, HitNormal, EndTrace, StartTrace)
+    {
+        if (IsFrobbable(target) && (target != CarriedDecoration))
+        {
+            if (target.IsA('ScriptedPawn'))
+            {
+                smallestTarget = target;
+                break;
+            }
+            else if (target.IsA('Mover') && bFirstTarget)
+            {
+                smallestTarget = target;
+                break;
+            }
+            else if (target.CollisionRadius < minSize)
+            {
+                minSize = target.CollisionRadius;
+                smallestTarget = target;
+                bFirstTarget = False;
+            }
+        }
+    }
 
-    Super.HighlightCenterObject();
+    // if we already have a frob target, and the player looks away such as that no item is being
+    // traced, we still wait for the full 100ms vanilla duration before clearing the frob
+    // target.
+    //
+    // note that this means we don't wait for the full 100ms vanilla duration if the player is
+    // rapidly changing frob target.
+    if ((FrobTime < 0.1) && (FrobTarget != None) && (smallestTarget == None))
```

Comment:
I'm honestly wondering if this should be higher than 0.1 lol

I'm gonna test the oceanlab shelves

Reply:
your change actually makes it a million times easier to grab those items, previously the timing could make it so the item never even got highlighted at all